src/compiler/scala/tools/nsc/Reporting.scala

```diff
@@ -544,16 +546,21 @@ object Reporting {
           }
         }
       } else if (s.startsWith("src=")) {
-        var pat = s.substring(4)
-        if (!pat.startsWith("/")) pat = "/" + pat
-        if (!pat.endsWith("$")) pat = pat + "$"
-        regex(pat).map(SourcePattern)
+        val arg = s.substring(4)
+        var pat = new mutable.StringBuilder()
+        if (rootDir.nonEmpty) pat += '^' ++= rootDir
```

Comment:
`rootDir` needs to be escaped with `Regex.quote`

Reply:
Thanks, done.